src/property.ts

```diff
@@ -913,14 +910,12 @@ function Property$subArrayEvents(obj: Entity, property: Property, array: Observa
 	});
 }
 
-function Property$getInitialValue(property: Property, obj: Entity): any {
+function Property$getInitialValue(property: Property): any {
 	// Constant
 	if (property.isConstant)
 		return typeof property.constant === "function" ? (property.constant = property.constant()) : property.constant;
 
-	var val = property.initializer
-		? property.initializer.call(obj)
-		: property.defaultValue;
```

Comment:
Don't call the initializer when getting the initial (default) value for a property, since a context-aware function could cause object creation, rule execution, etc. which could result in infinite "recursion" in the case of bi-directional relationships between objects and properties. Instead, the property will be given a default value, and then the initializer will be used to set or update the value (in `Property$ensureInited`).